bytecomp/lambda.mli

```diff
@@ -152,6 +152,9 @@ and comparison =
 and array_kind =
     Pgenarray | Paddrarray | Pintarray | Pfloatarray
 
+and block_kind =
+    Pgenblock | Pfloatblock | Pboxedintblock of boxed_integer
```

Comment:
The name of the type and constructor is quite misleading as this is applied to things that are not blocs